staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -295,45 +416,28 @@ func (cnc *CloudNodeController) updateNodeAddress(ctx context.Context, node *v1.
 // in a retry-if-conflict loop.
 type nodeModifier func(*v1.Node)
 
-func (cnc *CloudNodeController) UpdateCloudNode(ctx context.Context, _, newObj interface{}) {
-	node, ok := newObj.(*v1.Node)
-	if !ok {
-		utilruntime.HandleError(fmt.Errorf("unexpected object type: %v", newObj))
-		return
-	}
-
+// SyncNode handles updating existing nodes registered with the cloud taint.
+func (cnc *CloudNodeController) SyncNode(ctx context.Context, node *v1.Node) {
+	klog.Infof("Syncing node %s", node.Name)
```

Comment:
@andrewsykim I think maybe we should keep passing Node object for `SyncNode()` but only passing nodeName for `initializeNode()` cuz we still need to get **cloudTaint** of the node and see if there is any error

Reply:
I think we can remove the cloud taint check here since we have one already on line 457

Reply:
With that said it seems like we can probably collapse `SyncNode` and `initializeNode` into a single method `syncNode`.

Reply:
If we remove checking the cloud taint here, we should check the cloud taint in `initializeNode` _before_ adding `NodeNetworkUnavailable` condition for GCE on line 435

Reply:
If we remove checking the cloud taint here, we should check the cloud taint in `initializeNode` _before_ adding `NodeNetworkUnavailable` condition for GCE on line 435

Reply:
agree, will collapse these two functions and also make corresponding changes